core/src/main/java/me/mykindos/betterpvp/core/items/listener/UuidListener.java

```diff
@@ -91,13 +95,33 @@ public class UuidListener implements Listener {
     private final Map<Player, Inventory> lastInventory = new HashMap<>();
     private final Map<Player, UUIDItem> lastHeldUUIDItem = new HashMap<>();
 
+    private final Map<UUID, Long> lastUUIDDropTime = new HashMap<>();
+
     @Inject
     public UuidListener(Core core, ItemHandler itemHandler, ClientManager clientManager) {
         this.core = core;
         this.itemHandler = itemHandler;
         this.clientManager = clientManager;
     }
 
+    @EventHandler(priority = EventPriority.HIGHEST)
+    public void onUUIDItemDrop(PlayerDropItemEvent event) {
+        if (event.isCancelled()) return;
+        Client client = clientManager.search().online(event.getPlayer());
+        if ((boolean) client.getProperty(ClientProperty.DROP_PROTECTION_ENABLED).orElse(false)) {
+            itemHandler.getUUIDItem(event.getItemDrop().getItemStack()).ifPresent(uuidItem -> {
+                if (!(lastUUIDDropTime.containsKey(uuidItem.getUuid()) && !UtilTime.elapsed(lastUUIDDropTime.get(uuidItem.getUuid()), UUIDdropTimeMs))) {
```

Comment:
if map doesnt contain uuid and time has elapsed?

Could probably clean this up a bit, not very readable

Reply:
If the map doesn't contain the UUID, then this is the first time (this session) that we have seen this legend, therefore it is impossible for time to have elasped.

I was trying to avoid unnecessary/nested if statements, but I can see it being confusing

Reply:
Cleaned up

Reply:
much better